wallet/wallet-rpc-client/src/handles_client/mod.rs

```diff
@@ -103,14 +105,15 @@ impl<N: NodeInterface + Clone + Send + Sync + 'static + Debug> WalletInterface
         path: PathBuf,
         store_seed_phrase: bool,
         mnemonic: Option<String>,
+        passphrase: Option<String>,
     ) -> Result<CreatedWallet, Self::Error> {
```

Comment:
This doesn't model the way the wallet work correctly. The case with mnemonic being None and passphrase being Some is possible and this leads to having to manually check this or expect on it, unnecessarily. This should be enforced at the compiler level. Despite being ugly, something like `Option<(String, Option<String>)>` is way more appropriate (with docs explaining it). A struct for this is also OK.

Reply:
I thought the user might want to have a random seed phrase, but have his own passphrase

Reply:
Actually you're right. This is a valid case. Please document the meaning of these options, if you haven't yet.